pkg/providers/ovirt/client/ovirt-client.go

```diff
@@ -0,0 +1,289 @@
+}
+
+func (client *richOvirtClient) populateCdRoms(vm *ovirtsdk.Vm) error {
+	if cdroms, ok := vm.Cdroms(); ok {
+		if _, ok := cdroms.Href(); ok {
+			followed, err := client.connection.FollowLink(cdroms)
+			if err != nil {
+				return err
+			}
+			vm.SetCdroms(followed.(*ovirtsdk.CdromSlice))
+		}
+	}
+	return nil
+}
+
+func (client *richOvirtClient) populateFloppies(vm *ovirtsdk.Vm) error {
+	if fdds, ok := vm.Floppies(); ok {
+		if _, ok := fdds.Href(); ok {
+			followed, err := client.connection.FollowLink(fdds)
+			if err != nil {
+				return err
+			}
+			vm.SetFloppies(followed.(*ovirtsdk.FloppySlice))
+		}
+	}
+	return nil
+}
+
+func (client *richOvirtClient) populateWatchdogs(vm *ovirtsdk.Vm) error {
+	if dogs, ok := vm.Watchdogs(); ok {
+		if _, ok := dogs.Href(); ok {
+
+			followed, err := client.connection.FollowLink(dogs)
+			if err != nil {
+				return err
+			}
+			vm.SetWatchdogs(followed.(*ovirtsdk.WatchdogSlice))
+		}
+	}
+	return nil
+}
+
+func (client *richOvirtClient) populateGrapthicsConsoles(vm *ovirtsdk.Vm) error {
+	if consoles, ok := vm.GraphicsConsoles(); ok {
+		if _, ok := consoles.Href(); ok {
+			followed, err := client.connection.FollowLink(consoles)
+			if err != nil {
+				return err
+			}
+			vm.SetGraphicsConsoles(followed.(*ovirtsdk.GraphicsConsoleSlice))
+		}
+	}
+	return nil
+}
+
+func (client *richOvirtClient) populateQuota(vm *ovirtsdk.Vm) error {
+	if quota, ok := vm.Quota(); ok {
+		if _, ok := quota.Href(); ok {
+			followed, err := client.connection.FollowLink(quota)
+			if err != nil {
+				return err
+			}
+			vm.SetQuota(followed.(*ovirtsdk.Quota))
+		}
+	}
+	return nil
+}
+
+func (client *richOvirtClient) populateNics(vm *ovirtsdk.Vm) error {
+	if nics, ok := vm.Nics(); ok {
+		if _, ok := nics.Href(); ok {
+			followed, err := client.connection.FollowLink(nics)
+			nics = followed.(*ovirtsdk.NicSlice)
+			if err != nil {
+				return err
+			}
+			for _, nic := range nics.Slice() {
+				if prof, ok := nic.VnicProfile(); ok {
+					if _, ok := prof.Href(); ok {
+						vnicFollowed, err := client.connection.FollowLink(prof)
+						if err != nil {
+							return err
+						}
+						vnic := vnicFollowed.(*ovirtsdk.VnicProfile)
+						if net, ok := vnic.Network(); ok {
+							if _, ok := net.Href(); ok {
+
+								network, err := client.connection.FollowLink(net)
+								if err != nil {
+									return err
+								}
+								vnic.SetNetwork(network.(*ovirtsdk.Network))
+							}
+						}
+						nic.SetVnicProfile(vnic)
+					}
+				}
+				if net, ok := nic.Network(); ok {
+					if _, ok := net.Href(); ok {
+
+						network, err := client.connection.FollowLink(net)
+						if err != nil {
+							return err
+						}
+						nic.SetNetwork(network.(*ovirtsdk.Network))
+					}
+				}
+			}
+			vm.SetNics(nics)
+		}
+	}
+	return nil
+}
+
+func (client *richOvirtClient) populateDiskAttachments(vm *ovirtsdk.Vm) error {
+	if das, ok := vm.DiskAttachments(); ok {
+		if _, ok := das.Href(); ok {
```

Comment:
This check is actually already part of FollowLink: https://github.com/oVirt/go-ovirt/blob/d30f754823a6f8ccd317a23e3ead24d24cbb03de/connection.go#L117

Reply:
Please have a look at my explanation: https://github.com/kubevirt/vm-import-operator/pull/11#discussion_r393285221